### PR TITLE
Weather-getter; Changes based on code review

### PR DESCRIPTION
**Pull request by ElhamBigdelii**

### CODE REVIEW

File: weather_getter/app.py [hunk 1]
```diff
@@ -2,22 +2,56 @@
...
+import sys
+import re
 
 
 city_name = input("Please Enter City Name:")
 
+if re.fullmatch(r'^[a-zA-Z\s]+$',city_name) == None:
+    print("The city name is invalid.")
+    sys.exit()
+
```
Comment (salehmontazeran) [inline weather_getter/app.py:12]: Try to raise exception instead of printing and exiting.

File: weather_getter/app.py [hunk 2]
```diff
@@ -2,22 +2,56 @@
...
+if location:
+    if hasattr(location,'latitude'):
+        latitude = location.latitude
+    else:
+        latitude = ""
+    
+    if hasattr(location,'longitude'):
+        longitude = location.longitude
+    else:
+        latitude = ""
```
Comment (salehmontazeran) [inline weather_getter/app.py]: What happened if latitude set to ""? Does open-meteo require floating point latitude or latitude?

File: weather_getter/app.py [hunk 3]
```diff
@@ -2,22 +2,56 @@
...
+with open(file_name,"w") as resultFile:
+    resultFile.write(json.dumps(response.json(),indent=4))
+
+# if response.json().get('current') is not None:
+# if 'current' in response.json()
+try:
+    current_temp = response.json()['current']['temperature_2m']
+    print(f"{city_name}'s current temp is: {current_temp}")
+except KeyError:
+    print(f"An Error occur during get response. Please try later. Error is {KeyError}")
```
Comment (salehmontazeran) [inline weather_getter/app.py]: When you want to handle rare errors, you should try to test them with manual error. The last part of your print could be wrong, KeyError in a class name, and there is no meaning in using it like that.

**Review by salehmontazeran — CHANGES_REQUESTED** [review]

File: weather_getter/app.py (hunk 2 above)
Comment (ElhamBigdelii) [reply weather_getter/app.py]: what do you mean?

**Pull request merged**